Signal/src/environment/NotificationsManager.m

```diff
@@ -215,7 +216,7 @@ - (NotificationType)notificationPreviewType
 
 - (void)presentNotification:(UILocalNotification *)notification identifier:(NSString *)identifier
 {
-    dispatch_async(dispatch_get_main_queue(), ^{
+    DispatchMainThreadSafe(^{
         // Replace any existing notification
         // e.g. when an "Incoming Call" notification gets replaced with a "Missed Call" notification.
         if (self.currentNotifications[identifier]) {
```

Comment:
The next line is the bug: we call `cancelNotificationWithIdentifier` which is async and which will always cancel the notification we're trying to present. Bug only affects iOS 9.  Looks like on iOS 10 `dispatch_async(dispatch_get_main_queue...` will execute synchronously if you're already on the main thread.  I thought that same behavior was in earlier versions of iOS; I was wrong.